litmus-portal/graphql-server/pkg/myhub/gitops/gitops.go

```diff
@@ -0,0 +1,289 @@
+package gitops
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/litmuschaos/litmus/litmus-portal/graphql-server/graph/model"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+
+const (
+	defaultBranch = "master"
+)
+
+//GitConfig ...
+type GitConfig struct {
+	UserName       string
+	RepositoryName string
+	RepositoryURL  string
+	RemoteName     string
+	LocalCommit    string
+	RemoteCommit   string
+	HubName        string
+	Branch         string
+}
+
+var (
+	repository  *git.Repository
+	workTree    *git.Worktree
+	plumbingRef *plumbing.Reference
+	status      *git.Status
+	err         error
+)
+
+const (
+	defaultPath = "/tmp/version/"
+)
+
+//GetClonePath is used to construct path for Repository.
+func GetClonePath(c GitConfig) string {
+	RepoPath := defaultPath + c.UserName + "/" + c.HubName + "/" + c.RepositoryName + "/" + c.Branch
+	return RepoPath
+}
+
+//GitConfigConstruct is used for constructing the gitconfig
+func GitConfigConstruct(repoData model.ChartsInput) GitConfig {
+	gitConfig := GitConfig{
+		UserName:       repoData.UserName,
+		RepositoryName: repoData.RepoName,
+		HubName:        repoData.HubName,
+		RepositoryURL:  "https://github.com/" + repoData.RepoOwner + "/" + repoData.RepoName,
```

Comment:
```suggestion
		RepositoryURL:  "https://github.com/" + repoData.RepoOwner + "/" + repoData.RepoName,
```
this should not be specific to github. ex- bitbucket/gitlab will not work here. Ask whole URL from the user